cpp/tokenizers.cc

```diff
@@ -85,14 +125,159 @@ Tokenizer Tokenizer::FromPath(const String& _path) {
         << "Consider converting `tokenizer.model` to `tokenizer.json` by compiling the model "
         << "with MLC again, or see if MLC's huggingface provides this file.";
     return Tokenizer(
-        tokenizers::Tokenizer::FromBlobSentencePiece(LoadBytesFromFile(sentencepiece.string())));
+        tokenizers::Tokenizer::FromBlobSentencePiece(LoadBytesFromFile(sentencepiece.string())),
+        info_value);
   }
   if (std::filesystem::exists(rwkvworld)) {
-    return Tokenizer(tokenizers::Tokenizer::FromBlobRWKVWorld(rwkvworld.string()));
+    return Tokenizer(tokenizers::Tokenizer::FromBlobRWKVWorld(rwkvworld.string()), info_value);
   }
   LOG(FATAL) << "Cannot find any tokenizer under: " << _path;
 }
 
+TokenizerInfo Tokenizer::DetectTokenizerInfo(const String& path_str) {
+  std::filesystem::path path(path_str.operator std::string());
+  CHECK(std::filesystem::exists(path)) << "Cannot find tokenizer via path: " << path_str;
+  if (!std::filesystem::is_directory(path)) {
+    path = path.parent_path();
+  }
+  path = path / "tokenizer.json";
+  if (!std::filesystem::exists(path)) {
+    LOG(WARNING) << "Tokenizer info is not detected as tokenizer.json is not found. The default "
+                 << "tokenizer info will be used.";
+    return TokenizerInfo();
+  }
+
+  std::string tokenizer_json = LoadBytesFromFile(path.string());
+  picojson::value v;
```

Comment:
Isolate out another function DetectTokenizerInfoFromTokenizerJSON, and call into that. This is needed for platforms that may not come with this